core/src/main/java/org/elasticsearch/cluster/ClusterChangedEvent.java

```diff
@@ -134,10 +142,18 @@ public boolean indexRoutingTableChanged(String index) {
         return deleted == null ? Collections.<String>emptyList() : deleted;
     }
 
+    /**
+     * Returns <code>true</code> iff the metadata for the cluster has changed between
```

Comment:
Done, I also added that note to every other reference equality check in the class